src/lava/magma/compiler/compiler.py

```diff
@@ -496,6 +554,26 @@ def _get_channel_type(src: ty.Type[AbstractProcessModel],
                 f"'({src.__name__}, {dst.__name__})' yet."
             )
 
+    @staticmethod
+    def _get_port_dtype(port: AbstractPort,
+                        proc_model: ty.Type[AbstractProcessModel]) -> type:
+        """Returns the type of a port, as specified in the corresponding
+         ProcessModel."""
+
+        # In-, Out-, Ref- and explicit VarPorts
+        if hasattr(proc_model, port.name):
```

Comment:
Hmm... that's inconsistent. Here you check for existence of an attribute, below you check for 'isinstance'.

I would just use an if/elif/else ladder and distinguish by type.

Reply:
I would still need to check if the attribute is there for I/O, Ref and explicit VarPorts. 
Would look like this:

if isinstance(port, InPort) or isinstance(port, OutPort) or isinstance(port, RefPort):
    if hasattr(proc_model, port.name):
        return getattr(..)
    else:
        raise ...
elif isinstance(ImplicitVarPort):
    return getattr(..)
elif isinstance(VarPort):
    if hasattr(...):
        return ...
    else:
        raise ...
else:
    raise

Reply:
True... We are expecting it to be there. But right now we only check for an equivalent between Proc and ProcModel in the Builder. 

Perhaps this motivates to pull the check that Proc and ProcModel have the same ports and vars out of the building and just make this a stanalone compiler routine. 

That actually makes a lot more sense because right now, we would have to repeat this validation in the CProcBuilder.

This should be done upfront, then we don't even have to do this check here but can rely on it being here.

I will file an issue on this.